test/core/crawlers/basic_browser_crawler.ts

```diff
@@ -1,15 +1,30 @@
-import type { PuppeteerPlugin } from '@crawlee/browser-pool';
-import type { PuppeteerCrawlerOptions, PuppeteerCrawlingContext, PuppeteerGoToOptions } from '@crawlee/puppeteer';
+import type { PuppeteerController, PuppeteerPlugin } from '@crawlee/browser-pool';
+import type {
+    BrowserCrawlerOptions,
+    BrowserCrawlingContext,
+    PuppeteerCrawlingContext,
+    PuppeteerGoToOptions,
+} from '@crawlee/puppeteer';
 import { BrowserCrawler } from '@crawlee/puppeteer';
-import type { HTTPResponse, LaunchOptions } from 'puppeteer';
+import type { HTTPResponse, LaunchOptions, Page } from 'puppeteer';
+
+export type TestCrawlingContext = BrowserCrawlingContext<Page, HTTPResponse, PuppeteerController>;
 
 export class BrowserCrawlerTest extends BrowserCrawler<
+    Page,
+    HTTPResponse,
+    PuppeteerController,
     { browserPlugins: [PuppeteerPlugin] },
     LaunchOptions,
-    PuppeteerCrawlingContext
+    TestCrawlingContext
 > {
-    constructor(options: Partial<PuppeteerCrawlerOptions> = {}) {
-        super(options as any);
+    constructor(
+        options: Partial<BrowserCrawlerOptions<Page, HTTPResponse, PuppeteerController, TestCrawlingContext>> = {},
+    ) {
+        super({
+            ...options,
+            contextPipelineBuilder: () => this.buildContextPipeline(),
```

Comment:
Is this a necessary step when subclassing `Crawler` instances now?

Reply:
Only `BrowserCrawler`, and that one is _special_ in many ways...